tests/test_localedata.py

```diff
@@ -94,3 +95,16 @@ def test_mixedcased_locale():
         locale_id = ''.join([
             methodcaller(random.choice(['lower', 'upper']))(c) for c in l])
         assert localedata.exists(locale_id)
+
+def test_pi_support_frozen():
+    sys._MEIPASS, sys.frozen = 'testdir', True
```

Comment:
This should be done with Pytest's `monkeypatch` fixture:

```python
def test_pi_support_frozen(monkeypatch):
    monkeypatch.setattr('sys', '_MEIPASS', 'testdir')
    monkeypatch.setattr('sys', 'frozen', True)
    assert localedata.get_base_dir() == 'testdir'
```

Monkeypatch will ensure cleanup occurs.